src/Analyzer.Core.UnitTests/TemplateAnalyzerTests.cs

```diff
@@ -115,26 +104,14 @@ public void AnalyzeTemplate_BicepTemplateIsInvalid_ThrowTemplateAnalyzerExceptio
             try
             {
                 File.WriteAllText(templateFilePath, invalidBicep);
-                templateAnalyzerWithPowerShell.AnalyzeTemplate(invalidBicep, templateFilePath: templateFilePath);
+                templateAnalyzerAllRules.AnalyzeTemplate(invalidBicep, templateFilePath: templateFilePath);
             }
             finally
             {
                 File.Delete(templateFilePath);
             }
         }
 
-        [TestMethod]
-        [ExpectedException(typeof(TemplateAnalyzerException))]
-        public void AnalyzeTemplate_MissingFilePathWithPowerShellOn_ThrowTemplateAnalyzerException()
```

Comment:
Why was this test removed?

Reply:
It was there to validate a check that doesn't exist anymore, the file path now is a required parameter of the function, it will always fail if it's not specified

Reply:
> it will always fail if it's not specified

I would argue that behavior should be validated with this test.

Reply:
Do we validate for all our functions that they fail when we don't pass a required argument? :P

Reply:
We should.  If it's required, we should validate that the code enforces the requirement.  Example tests for LeafExpression: https://github.com/Azure/template-analyzer/blob/development/src/Analyzer.JsonRuleEngine.UnitTests/LeafExpressionTests.cs#L144#L156

Reply:
Put another way, if it's required, but there's a logical path in code where that requirement is somehow bypassed, there's a bug.  So the requirement should be tested.

Reply:
I think we are not understanding each other, that test doesn't even compile anymore, because the function call is missing a parameter

Reply:
I think we are not understanding each other, that test doesn't even compile anymore, because the function call is missing a parameter. It doesn't make sense to test what happens when the parameter is missing, because it doesn't compile when that happens

Reply:
Previously it was set to null by default, so it could be "left out".  The test should pass `null` as the value and expect a failure.

Reply:
Maybe "missing" vs. "null" is the confusing piece.  It should validate a failure from passing null as the file path.